app/models/versioned/image/revision.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Versioned
+  # An image revision represents an edit of a particular image, it's data is
+  # stored across two associations: Image::FileRevisiona and
+  # Image::MetadataRevision. The file revision stores data that when changed
+  # causes new files to be uploaded for asset manager whereas metadata stores
+  # data that is doesn't require changes to the asset manager files.
+  #
+  # This is an immutable model
+  class Image::Revision < ApplicationRecord
+    self.table_name = "versioned_image_revisions"
+
+    COMPARISON_IGNORE_FIELDS = %w[id created_at created_by_id].freeze
+
+    # rubocop:disable Rails/InverseOf
+    belongs_to :created_by,
+               class_name: "User",
+               optional: true,
+               foreign_key: :created_by_id
+    # rubocop:enable Rails/InverseOf
+
+    has_and_belongs_to_many :revisions,
+                            foreign_key: "image_revision_id",
+                            class_name: "Versioned::Revision",
+                            join_table: "versioned_revision_image_revisions"
+
+    belongs_to :image,
+               class_name: "Versioned::Image",
+               foreign_key: :image_id,
+               inverse_of: :image_revisions
+
+    belongs_to :file_revision,
+               class_name: "Versioned::Image::FileRevision",
+               foreign_key: :file_revision_id,
+               inverse_of: :revisions
+
+    belongs_to :metadata_revision,
+               class_name: "Versioned::Image::MetadataRevision",
+               foreign_key: :metadata_revision_id,
+               inverse_of: :revisions
+
+    delegate :alt_text,
+             :caption,
+             :credit,
+             to: :metadata_revision
+
+    delegate_missing_to :file_revision
```

Comment:
Can we specify exactly what we need to delegate?

Reply:
yes - but it is quite a lot may end up quite verbose.

Reply:
True, but I think this is another scenario that we could just pass the specific revision around e.g. crop.html.erb. At least expanding it out makes it more obvious to change, without the mystery of delegate_missing.